tests/unit/src/kakarot/accounts/contract/test_contract_account.py

```diff
@@ -90,3 +90,29 @@ async def test_should_give_inifinite_allowance_to_kakarot(
                 )
                 == "Uint256(low=340282366920938463463374607431768211455, high=340282366920938463463374607431768211455)"
             )
+
+    """
+    class TestNonce:
+        async def test_should_increment_nonce(
+            self, contract_account: StarknetContract, kakarot
+        ):
+            # Get current contract account nonce
+            initial_nonce = await contract_account.get_nonce.call()
+
+            # Store bytecode that includes CREATE opcode
+            contract_account.write_bytecode([0]).execute(
+                caller_address=kakarot.contract_address
+            )
+
+            # Run bytecode
+            kakarot.execute_at_address(nonce: felt, bytecode_len: felt, bytecode: felt*)
+
+            # Get new nonce
+            assert initial_nonce == await contract_account.get_nonce().call()
+
+
+        async def test_should_influence_address_generation(
+            self, contract_account: StarknetContract, kakarot
+        ):
+            x = 1
+    """
```

Comment:
@ClementWalter 
I'm looking to create some tests to see if the nonces are handled correctly.

What I'm going for here is more an integration test, right?
What scope do you think would be fitting?

Reply:
so the only way for a contract_account to have its nonce increased is to have a CREATEright?

At the unit test level you may only want to test that `get_nonce` returns the nonce and `increment_nonce` increment it.

At the integration test level, you may want to test that when deploying a contract with CREATE the nonce increases